Userland/Utilities/imgcmp.cpp

```diff
@@ -20,10 +22,49 @@ static ErrorOr<RefPtr<Gfx::Bitmap>> load_image(StringView path)
     return TRY(decoder->frame(0)).image;
 }
 
+static ErrorOr<void> save_image(NonnullRefPtr<Gfx::Bitmap> bitmap, StringView out_path)
+{
+    if (!out_path.ends_with(".png"sv, CaseSensitivity::CaseInsensitive) && !out_path.ends_with(".webp"sv, CaseSensitivity::CaseInsensitive))
+        return Error::from_string_view("can only save to .png and .webp files"sv);
+
+    auto output_stream = TRY(Core::File::open(out_path, Core::File::OpenMode::Write));
+    auto buffered_stream = TRY(Core::OutputBufferedFile::create(move(output_stream)));
+
+    if (out_path.ends_with(".png"sv, CaseSensitivity::CaseInsensitive))
+        return Gfx::PNGWriter::encode(*buffered_stream, *bitmap);
+
+    VERIFY(out_path.ends_with(".webp"sv, CaseSensitivity::CaseInsensitive));
+    return Gfx::WebPWriter::encode(*buffered_stream, *bitmap);
+}
+
+static ErrorOr<NonnullRefPtr<Gfx::Bitmap>> make_diff_image(NonnullRefPtr<Gfx::Bitmap> first_image, NonnullRefPtr<Gfx::Bitmap> second_image)
+{
+    VERIFY(first_image->size() == second_image->size());
+
+    auto diff_image = TRY(Gfx::Bitmap::create(Gfx::BitmapFormat::BGRA8888, first_image->size()));
+
+    for (int y = 0; y < first_image->height(); ++y) {
+        for (int x = 0; x < first_image->width(); ++x) {
+            auto first_pixel = first_image->get_pixel<Gfx::StorageFormat::BGRA8888>(x, y);
+            auto second_pixel = second_image->get_pixel<Gfx::StorageFormat::BGRA8888>(x, y);
+            if (first_pixel == second_pixel) {
+                diff_image->set_pixel(x, y, first_pixel.interpolate(Gfx::Color::White, 0.5f));
+            } else {
+                diff_image->set_pixel(x, y, Gfx::Color::from_rgb(0xff0000));
```

Comment:
That should work: `diff_image->set_pixel(x, y, { Color::Red });`

Reply:
Don't even need the braces. Thanks, much nicer, done :)